### PR TITLE
Refactor with_analytics function into a decorator

### LINKED ISSUE #393: Refactor `with_analytics` into a decorator
### Is your feature request related to a specific problem?

Many CLI handler functions are wrapped in the `with_analytics` helper.

### Describe the solution you'd like

The `with_analytics` helper could more elegantly exist as a decorator. Refactor the function such that it supports being called in this manner.

### Describe alternatives you've considered, if any

We can make no changes - the code works as-is.

### PR DESCRIPTION
**Pull request by earmenda**
Closes #393

### Code Changes

* [x] Create `with_analytics_decorator` decorator
* [x] Apply `@with_analytics_decorator` to all cli functions
* [x] Delete old `with_analytics` function 

### Steps to Confirm

* [x] Manual test of cli execution and event publishing

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x]  Changelog Updated

### Description Of Changes

As is, the changes should equate to the same behavior but using the new decorator now. This could still use some tests, last thing im looking into.

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@earmenda is this still in draft or ready for review? This is the final PR for 1.6

**Comment by ThomasLaPiana** [comment]
please merge main to this so we can make sure to update the changelog as well

File: src/fidesctl/cli/utils.py [hunk 1]
```diff
@@ -185,7 +186,8 @@ def with_analytics(ctx: click.Context, command_handler: Callable, **kwargs: Dict
             except AnalyticsError:
                 pass  # cli analytics should fail silently
 
-def with_analytics_decorator(ctx: click.Context, command_handler: Callable, **kwargs: Dict) -> Any:  # type: ignore
+
+def with_analytics_decorator(func) -> Any:  # type: ignore
```
Comment (brentonmallen1) [inline src/fidesctl/cli/utils.py:190]: I think return type could be `Callable` here?

Comment (earmenda) [reply src/fidesctl/cli/utils.py:190]: yeah true, will update the function doc also.

**Comment by SteveDMurphy** [comment]
@earmenda the only thing I am seeing so far is when testing I am getting an error: `The fideslog API responded with an error: 422 Unprocessable Entity`

Were you able to send events with this earlier today by chance? It looks like this may be down to `fideslog` instead but wanted to get a message out here quick while investigating

**Comment by earmenda** [comment]
> @earmenda the only thing I am seeing so far is when testing I am getting an error: `The fideslog API responded with an error: 422 Unprocessable Entity`
> 
> Were you able to send events with this earlier today by chance? It looks like this may be down to `fideslog` instead but wanted to get a message out here quick while investigating

Hmmm where are you seeing that error? I added some logging around `ctx.meta["ANALYTICS_CLIENT"].send(event)` and I don't see it throw an error when I test my api calls.

**Comment by SteveDMurphy** [comment]
> > @earmenda the only thing I am seeing so far is when testing I am getting an error: `The fideslog API responded with an error: 422 Unprocessable Entity`
> > Were you able to send events with this earlier today by chance? It looks like this may be down to `fideslog` instead but wanted to get a message out here quick while investigating
> 
> Hmmm where are you seeing that error? I added some logging around `ctx.meta["ANALYTICS_CLIENT"].send(event)` and I don't see it throw an error when I test my api calls.

I moved `ctx.meta["ANALYTICS_CLIENT"].send(event)` outside of the try to see the error returned (since it was silent with no event being sent). Can see that yours was just sent now however so this must be a me thing, thanks for double checking!

File: src/fidesctl/cli/utils.py [hunk 2]
```diff
@@ -144,43 +145,45 @@ def send_init_analytics(opt_out: bool, config_path: str, executed_at: datetime)
...
-    :param command_handler: The handler function defining the evaluation logic for the analyzed command
-    :param **kwargs: Any arguments that must be passed to the `command_handler` function
+    :param func: function to be wrapped by decorator
     """
 
-    command = " ".join(filter(None, [ctx.info_name, ctx.invoked_subcommand]))
-    error = None
-    executed_at = datetime.now(timezone.utc)
-    status_code = 0
+    def new_func(ctx: click.Context, *args, **kwargs) -> Any:  # type: ignore
```
Comment (SteveDMurphy) [inline src/fidesctl/cli/utils.py:157]: Opinion here but thinking there may be something more descriptive we can use besides `new_func`?

**Review by SteveDMurphy — APPROVED** [review]
Thanks for validating my earlier issue @earmenda ! I still need to figure out why I was getting an error but this all looks good to me and can sort that out separately. Added one nit around naming is all but nothing to hold this up

File: src/fidesctl/cli/utils.py (hunk 2 above)
Comment (earmenda) [reply src/fidesctl/cli/utils.py]: sure, renamed it to `wrapper_func`. Lmk if that's not okay!

Comment (SteveDMurphy) [reply src/fidesctl/cli/utils.py]: sounds good, thanks!

**Review by PSalant726 — APPROVED** [review]
@earmenda Nice change!

**Review by ThomasLaPiana — APPROVED** [review]
Excellent change, super pythonic!

**Pull request merged**